protocols/identify/src/behaviour.rs

```diff
@@ -294,8 +300,27 @@ impl NetworkBehaviour for Behaviour {
                 let observed = info.observed_addr.clone();
                 self.events
                     .push_back(ToSwarm::GenerateEvent(Event::Received { peer_id, info }));
-                self.events
-                    .push_back(ToSwarm::NewExternalAddrCandidate(observed));
+
+                match self.our_observed_addresses.entry(id) {
+                    Entry::Vacant(not_yet_observed) => {
+                        not_yet_observed.insert(observed.clone());
+                        self.events
+                            .push_back(ToSwarm::NewExternalAddrCandidate(observed));
+                    }
+                    Entry::Occupied(already_observed) if already_observed.get() == &observed => {
+                        // No-op, we already observed this address.
+                    }
+                    Entry::Occupied(mut already_observed) => {
+                        log::info!(
+                            "Our observed address on connection {id} changed from {} to {observed}",
+                            already_observed.get()
+                        );
+
+                        *already_observed.get_mut() = observed.clone();
+                        self.events
+                            .push_back(ToSwarm::NewExternalAddrCandidate(observed));
+                    }
```

Comment:
This is the "roaming connection" usecase. Not sure if it is worth supporting?

Reply:
I don't think it adds much noise. In other words, I am fine keeping it here.